lib/datasource/docker/index.ts

```diff
@@ -420,33 +424,39 @@ async function getTags(
   }
 }
 
+export function getConfigResponseBeforeRedirectHook(options: any): void {
+  if (options.search?.includes('X-Amz-Algorithm')) {
+    // if there is no port in the redirect URL string, then delete it from the redirect options.
+    // This can be evaluated for removal after upgrading to Got v10
+    const portInUrl = options.href.split('/')[2].split(':')[1];
+    if (!portInUrl) {
+      // eslint-disable-next-line no-param-reassign
+      delete options.port; // Redirect will instead use 80 or 443 for HTTP or HTTPS respectively
+    }
+
+    // docker registry is hosted on amazon, redirect url includes authentication.
+    // eslint-disable-next-line no-param-reassign
+    delete options.headers.authorization;
+  }
+
+  if (
+    options.href?.includes('blob.core.windows.net') &&
```

Comment:
```suggestion
    options.href.includes('blob.core.windows.net') &&
```

Maybe null check not required, as href is alwys set?

Reply:
But also no major downside, right?